coderd/coderd.go

```diff
@@ -152,7 +153,12 @@ type Options struct {
 	MetricsCacheRefreshInterval time.Duration
 	AgentStatsRefreshInterval   time.Duration
 	DeploymentValues            *codersdk.DeploymentValues
-	UpdateCheckOptions          *updatecheck.Options // Set non-nil to enable update checking.
+	// DeploymentOptions do contain the copy of DeploymentValues, but contain
+	// contextual information about how the values were set.
+	// Do not use DeploymentOptions to retrieve values, use DeploymentValues instead.
```

Comment:
Why not just strip all values from `DeploymentOptions` then? Too many allocations?

Reply:
I wanted to go this route, but the UI grabs the value from the option. Values are returned in the parsed form, and the option form: 
https://github.com/coder/coder/blob/b1058a64bb926dbd9cd63c69661eacd0a9aac173/coderd/deployment.go#L34-L37

There is no way to associate the values between the two, so the value needs to be present on the Option for the UI to know.

I feel like this could use a bit of a refactor, but it is working and not really that important.